src/main/java/ru/l1ttleO/boyfriend/settings/Settings.java

```diff
@@ -0,0 +1,255 @@
+            this.save(key);
+        }
+
+        public @Nullable String getString(final K key) {
+            final V value = this.get(key);
+            return value == null ? null : value.toString();
+        }
+
+        public abstract V putFromString(final K key, final @NotNull String str);
+
+        public @Nullable String formatted(final K key) {
+            final String str = this.getString(key);
+            return str == null ? null : "`" + str + "`";
+        }
+
+        public V setFormatted(final K key, final String str) throws IllegalArgumentException {
+            final V value = this.putFromString(key, str);
+            this.save(key);
+            return value;
+        }
+    }
+
+    public static class StringEntry<K> extends Entry<K, String> {
+        public StringEntry(final @NotNull Function<K, Settings> keyGetter, final @NotNull String name, final String defaultValue) {
+            super(keyGetter, name, "string", defaultValue);
+        }
+
+        @Override
+        public String putFromString(final K key, final String str) throws IllegalArgumentException {
+            put(key, str);
+            return str;
+        }
+    }
+
+    public static class BotLocaleEntry<K> extends Entry<K, BotLocale> {
+        public BotLocaleEntry(final @NotNull Function<K, Settings> keyGetter, final @NotNull String name, final BotLocale defaultValue) {
+            super(keyGetter, name, "locale", defaultValue);
+        }
+
+        @Override
+        public String getString(final K key) {
+            return this.get(key).name().toLowerCase();
+        }
+
+        @Override
+        public BotLocale putFromString(final K key, final String str) throws IllegalArgumentException {
+            final BotLocale locale;
+            try {
+                locale = BotLocale.valueOf(str.toUpperCase());
+            } catch (final IllegalArgumentException e) {
+                throw new IllegalArgumentException("Locale is unavailable: " + str);
+            }
+            put(key, locale);
+            return locale;
+        }
+
+        @Override
+        public BotLocale setFormatted(final K key, final @NotNull String str) throws IllegalArgumentException {
+            final BotLocale locale = BotLocale.valueOf(str.toUpperCase());
+            put(key, locale);
+            save(key);
+            return locale;
+        }
+    }
+
+    public static class TextChannelEntry<K> extends Entry<K, TextChannel> {
+        public TextChannelEntry(final @NotNull Function<K, Settings> keyGetter, final @NotNull String name) {
+            super(keyGetter, name, "channel", null);
+        }
+
+        @Override
+        public @Nullable String getString(final K key) {
+            final TextChannel channel = this.get(key);
+            return channel == null ? null : channel.getId();
+        }
+
+        @Override
+        public TextChannel putFromString(final K key, final String str) throws IllegalArgumentException {
```

Comment:
```suggestion
        public TextChannel putFromString(final K key, final @NotNull String str) throws IllegalArgumentException {
```
`Overriden method parameters are not annotated` [intellij] [3/4]